src/main/java/seedu/apollo/command/module/ListModuleWithLessonCommand.java

```diff
@@ -0,0 +1,234 @@
+                module.setTimetable(module1.getModuleTimetable());
+                break;
+            }
+            index++;
+        }
+        module.setTimetable(moduleList.get(index).getModuleTimetable());
+        if (module.getModuleTimetable().isEmpty()) {
+            throw new LessonTypeNotAddedException();
+        }
+
+    }
+
+    /**
+     * Returns the available lesson type of the module.
+     *
+     * @param module The module being checked.
+     * @return The lesson types available for this module.
+     */
+    private ArrayList<LessonType> getLessonTypes(Module module) {
+        ArrayList<LessonType> lessonTypes = new ArrayList<>();
+        for (Timetable timetable : module.getModuleTimetable()) {
+            LessonType lessonType = determineLessonType(timetable.getLessonType());
+            if (!lessonTypes.contains(lessonType) && lessonType != null) {
+                lessonTypes.add(lessonType);
+            }
+        }
+        return lessonTypes;
+    }
+
+    /**
+     * Sorts the timetable by lesson type, class number and class time .
+     *
+     * @param copyList The list of lessons to be sorted.
+     * @return The sorted list of lessons.
+     */
+    private ArrayList<Timetable> sortTimetable(ArrayList<Timetable> copyList) {
+        Comparator<Timetable> compareByLessonType = Comparator.comparing(Timetable::getLessonType);
+        Comparator<Timetable> compareByClassNumber = Comparator.comparing(Timetable::getClassnumber);
+        Comparator<Timetable> compareAll = compareByLessonType.thenComparing(compareByClassNumber);
+        ArrayList<Timetable> parseList =
+                copyList.stream().sorted(compareAll).collect(Collectors.toCollection(ArrayList::new));
+        return parseList;
+    }
+
+    /**
+     * Checks if the module is in the module list.
+     *
+     * @param moduleList The list of modules.
+     * @param module     The module to be checked.
+     * @return True if the module is in the list of modules.
+     */
+    public boolean isInModuleList(ModuleList moduleList, Module module) {
+        for (Module moduleCheck : moduleList) {
+            if (moduleCheck.getCode().equals(module.getCode())) {
+                return true;
+            }
+        }
+        return false;
+    }
+
+    /**
+     * Handles the command when user wants to see the timetable added of a specific lesson type.
+     *
+      * @param ui The Ui object to print the timetable.
+     * @throws IllegalCommandException If the command is invalid.
+     */
+    private void handleMultiCommand(Ui ui) throws IllegalCommandException {
+
+        String type = args[1];
+
+        LessonType lessonType = getCommand(type);
+        if (lessonType == null) {
+            throw new IllegalCommandException();
+        }
+
+        ArrayList<Timetable> timetableList = new ArrayList<>(module.getModuleTimetable());
+
+        ArrayList<Timetable> copyList = new ArrayList<>();
```

Comment:
naming of this can be a bit clearer what its for, instead of copyList maybe addedTimetables??? idk u can probably come up w a better name than me HAHA